### DISCUSSION TITLE
Target Find Expression display problem in Relation Based Edge

### DISCUSSION THREAD

**Discussion by pelva** [eclipse-sirius/sirius-desktop#334] [General]
I have a question: Does Target Find Expression allow entering multi-line expressions?
Because when editing a multi-line aql expression, the edit box will display...

![image](https://github.com/eclipse-sirius/sirius-desktop/assets/135562884/133417da-e6bf-48b0-a56e-b5e575c892f8)

![image](https://github.com/eclipse-sirius/sirius-desktop/assets/135562884/18e8df69-afa0-4b52-acc2-276007cf18c6)
![image](https://github.com/eclipse-sirius/sirius-desktop/assets/135562884/4bc6fb08-e194-450d-aa4c-462c67f4cd18)

In the single-line input box, there is no problem in displaying like this, but if it also displays in the pop-up multi-line editor, is there a problem?
Is this a bug or a usage problem on my part?

Sirius version is 7.0.1

**Comment by lredor** [comment]
Hi pelva,
As said in the [documentation](https://eclipse.dev/sirius/doc/specifier/general/Specifying_Viewpoints.html#interpreted_expressions), `Also, by clicking on the button next to the text field, you can open a dialog with a bigger text area (providing auto-completion too) to type your expression more comfortably. Note that you can add line feeds or tabulations if you want, since they are removed when clicking on the “OK” button.`. But it seems that there is a bug. The carriage return is stored in the VSM file (`&#xD;`) and is wrongly interpreted then.

**Reply by pelva** [reply]
Yes,I find the  (&#xD) 
Now I have to write all statements in one line which makes it very less readable

**Comment by lredor** [comment]
As additional information, after some research, this problem has already been detected in the past (bugzilla [480981](https://bugs.eclipse.org/bugs/show_bug.cgi?id=480981)). This topic had not been prioritized at the time and the "..." button allowing multi-line editing had been removed (in order to no longer allow the "corrupted" functionality). This change was made in Sirius 3.1.0 in 2015. However in 2018, starting with Sirius 6.1.0, the button "wrongly" reappeared. So actually, currently, as you said, you only could edit expression on one line.
